lib/ddtrace/contrib/grape/endpoint.rb

```diff
@@ -196,6 +204,54 @@ def analytics_sample_rate
             datadog_configuration[:analytics_sample_rate]
           end
 
+          def error_responses
+            return datadog_configuration[:error_responses] if datadog_configuration[:error_responses].is_a?(String) && !status.nil?
+            datadog_configuration[:error_responses].join(',') if status.is_a?(Array) && !status.empty?
+          end
+
+          def handle_statuses
+            if error_responses
+              error_responses.gsub(/\s+/, '').split(',').select do |code|
+                if !code.to_s.match(/^\d{3}(?:-\d{3})?(?:,\d{3}(?:-\d{3})?)*$/)
```

Comment:
Let's make abstract as much of this regex stuff into constants as we can, it should help perf, see an [example here](https://github.com/DataDog/dd-trace-rb/blob/0574335134aad3eb57cb6515f73943bc1afd6b60/lib/ddtrace/runtime/container.rb#L8-L12)